broker.go

```diff
@@ -46,6 +46,24 @@ type Broker struct {
 	brokerResponseSize     metrics.Histogram
 }
 
+// SaslMechanism specifies the SASL mechanism the client uses to authenticate with the broker
+type SaslMechanism string
+
+const (
+	// SaslTypeOAuth represents the SASL/OAUTHBEARER mechanism (Kafka 2.0.0+)
+	SaslTypeOAuth = "OAUTHBEARER"
+	// SaslTypePlaintext represents the SASL/PLAIN mechanism
+	SaslTypePlaintext = "PLAIN"
+)
+
+// OAuthBearerTokenProvider is an interface that encapsualtes bearer token creation in an
+// unopinionated way. Users are free to decide how to construct bearer tokens and how often
+// they are be refreshed.
+type OAuthBearerTokenProvider interface {
+	// Return a valid bearer token.
+	Token() []byte
```

Comment:
I'd recommend adding some more control over this definition by making it:

```go
Token(ctx context.Context) ([]byte, error)
```

The context will allow more control over timeouts and cancellation while fetching a token. The error also will allow implementors to alert the system if there were problems fetching/creating a new token. This is similar to [oauth2.TokenSource](https://godoc.org/golang.org/x/oauth2#TokenSource)

Reply:
Changed the interface to return an error. Unless someone feels strongly about this, I am going to hold off on the `context.Context` param for now because I think it will keep the configuration simpler (passing a user-configured context object to a user-defined callback seems redundant to me).